californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -848,7 +848,7 @@ private Request ifNoneMatch(Request request) {
 	 * @param handler the Response handler
 	 * @return the CoAP observe relation
 	 */
-	private CoapObserveRelation observeAndWait(Request request, CoapHandler handler) {
+	public CoapObserveRelation observeAndWait(Request request, CoapHandler handler) {
 		Endpoint outEndpoint = getEffectiveEndpoint(request);
```

Comment:
Since this is now public, we will need to make sure that the observe option is set.